### DISCUSSION TITLE
Question about task-based fMRI data

### DISCUSSION THREAD

**Discussion by frankgarcea** [abhogal-lab/seeVR#15] [Ideas]
Hi @abhogal-lab 

I'm a new user and have a few questions about the seeVR toolbox. I collect breath holding fMRI data in addition to resting and task fMRI data in participants with brain tumor or stroke. I'd like to get a hemodynamic lag map prior to modeling the BOLD data or computing functional connectivity between ROIs.

Can the seeVR toolbox be applied to task based fMRI data? I see you have documented the process of analyzing a breath holding task using a probe as input. To keep it simple: Imagine if participants hold their breath for 10 seconds every 30 seconds. I can create a timeseries of breath holding (0 = breathe normally, 1 = hold breath) for each TR in my task. This is straightforward.

Presumably I could do the same thing for a picture viewing fMRI task. By modeling when the participant saw a picture (1) or stared at a fixation cross (0), I can create a probe for each run to give to the lagCVR call. Does that make sense?

Is there a way to apply the lagCVR call to resting fMRI data? Would the probe just be a vector of 1s?

Thanks in advance,
FG

**Comment by abhogal-lab** [comment]
Hi Frank,

First, thanks for your interest in the toolbox. I’m still developing it but I hope there are some useful things in there for people like yourself. 

I can think a few ways to skin this cat using the toolbox… 

1)	I’ve recently added a recursive lag function based on the work of Dr. Toshihiko Aso that was originally developed for RS-fMRI to map the perfusion lag structure. We’ve found that this method provides better r values and in general seemingly more realistic lag values that simple cross correlation. Dr. Aso’s code also attempts to remove this lag structure under the pretence that it improve functional connectivity metrics (i.e. to focus more on neuronal responses). I haven’t yet implemented the removal process (ill spare the details here). See: https://github.com/aso-toshihiko/BOLDLagMapping_Deperfusioning

2)	The glmCVRidx function calculates CVR based on this method: https://pubs.rsna.org/doi/full/10.1148/radiol.2021203568 by Peiying. The nice thing about this function is that it will output the bandpassed BOLD and reference signals. You can use these as inputs to the lagCVR function which will subsequently do either a cross correlation based lag analysis or a lagged-glm analysis depending on the options you choose. This function (as with the recursive lag function) also has the option to do regressor optimization. So it will take a reference signal and search through a reference mask (i.e. GM) and look for highly correlated voxels in an iterative process. In the end you get a BOLD signal regressor which can be used to perform xcorr or glm-based lag estimation – based on the RAPIDtide approach by Frederick et al. This is how it is done in the breath holding tutorial on www.seeVR.nl. 

You’re idea could also work. In this case, you would make your regressor from 0-1 and then either do a straight xcorr, or allow it to use that as a seed for regressor optimization. The nice thing with the regressor optimization is that you can specify a mask. For example, in our stroke data, we are now using the cerebellum as the seed since presumably it is hemodynamically unaffected by the stroke lesion.

Best,
Alex
